src/components/Modal/Modal.tsx

```diff
@@ -86,6 +82,96 @@ type ModalProps = ModalContentProps & {
   modalContainerClassName?: string;
 };
 
+type ModalTitleProps = Omit<ExtractProps<typeof Heading>, 'size'> & {
+  /**
+   * Text for the modal title.
+   */
+  children: ReactNode;
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
```

Comment:
There are gonna be some opportunities to consolidate and share common types once we get all of these reduced down